### DISCUSSION TITLE
Duplicates search options

### DISCUSSION THREAD

**Discussion by gnc-tibo** [imapsync/imapsync#462] [Q&A]
Hello,

I tried using these options to migrate one large mailbox to another by cleaning up the many duplicates.

```
CALL .\imapsync.exe  ^
  --host1 outlook.office365.com --user1 "archive.username@domain.net" --authuser1 "administrator@domain.net"  --oauthaccesstoken1 tokens\oauth2_tokens_administrator@domain.net.txt ^
  --host2 outlook.office365.com  --user2 "username@domain.net"  --authuser2 "administrator@domain.net" --oauthaccesstoken2 tokens\oauth2_tokens_administrator@domain.net.txt  ^
  --delete2duplicates --skipemptyfolders --skipcrossduplicates
```

But some emails are missing, for example :
```
Folder 1391/1467 [FOLDERNAME]                      -> [FOLDERNAME]                     
Host1: folder [FOLDERNAME] has 92 messages in total (mentioned by SELECT)
Host2: folder [FOLDERNAME] has 12 messages in total (mentioned by SELECT)
Host2: folder [FOLDERNAME] permanentflags: \Seen \Answered \Flagged \Deleted \Draft $MDNSent
Host1: folder [FOLDERNAME] considering 92 messages
Host2: folder [FOLDERNAME] considering 12 messages
Host1: folder [FOLDERNAME] selected 92 messages, duplicates 40
Host2: folder [FOLDERNAME] selected 12 messages, duplicates 0
```
Host 2 should have 52 mails, only 12 have been transferred. Using `--skipcrossduplicates`, I searched in the other folders to see if any missing emails would be there, but that doesn't seem to be the case.

What could be the explanations? Would there be a contraindication to using together `--skipcrossduplicates` and `--delete2duplicates` ?

Thanks !

**Comment by gilleslamiral** [comment]
Hello,

What gives --debugcrossduplicates --delete2
What gives --folder FOLDERNAME --delete2

**Comment by gnc-tibo** [comment]
Hello,

Cannot use `--delete2` option because user is already using his new mailbox ☹️

I will eventually see about testing on a new mailbox.

**Comment by gilleslamiral** [comment]
Test it without --delete2 or with -dry

**Comment by gnc-tibo** [comment]
Hello,

Sorry for not getting back to you sooner!
`--debugcrossduplicates` was the solution and allowed to identify duplicates that Outlook search was not locating.

Many thanks!

**Comment by gilleslamiral** [comment]
Great!
I close the ticket.
